lib/Sema/SemaBounds.cpp

```diff
@@ -613,14 +614,21 @@ namespace {
             llvm_unreachable("unexpected cast failure");
             return CreateBoundsNone();
           }
+
           Expr *subExpr = CE->getSubExpr();
           BoundsExpr *Bounds = CE->getBoundsExpr();
-          QualType QT = subExpr->getType();
-	  Expr *Base = subExpr;
-          if (subExpr->getStmtClass() != clang::Stmt::CallExprClass)
-	    Base=CreateImplicitCast(QT, CastKind::CK_LValueToRValue, subExpr);
+
```

Comment:
You don't need lines 621-629.  The bounds are just the declared bounds of the bounds-cast expression (the value of Bounds).